### PR TITLE
Pytests refactor

### PR DESCRIPTION
**Pull request by Melvie**
Refactored test in python plugin lib and set up to run with travis as per Issue #249

### CODE REVIEW

File: python/xi_plugin.egg-info/PKG-INFO [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Metadata-Version: 1.0
```
Comment (cmyr) [inline python/xi_plugin.egg-info/PKG-INFO]: Is including the .egg necessary?

File: rust/Cargo.lock [hunk 2]
```diff
@@ -37,6 +37,11 @@ version = "0.8.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
```
Comment (cmyr) [inline rust/Cargo.lock]: can we exclude the Cargo.lock from this PR?

Comment (Melvie) [inline rust/Cargo.lock]: I'm super new to rust, but from the reading I've done it seems like we can exclude the cargo.lock file for this PR.

File: python/xi_plugin.egg-info/PKG-INFO (hunk 1 above)
Comment (Melvie) [inline python/xi_plugin.egg-info/PKG-INFO]: I branched and tested pushing with travis without the egg and it fails. So, I created a setup.py file which will build the egg when the tests are run. So, no  including the .egg isn't necessary.

I did find though that python wasn't building on the osx worker though so I might close this request and open a new one when I figure out that

File: rust/Cargo.lock (hunk 2 above)
Comment (cmyr) [inline rust/Cargo.lock]: Yes, my phrasing was maybe a bit misleading. I meant more like, "can we _please_ exclude the Cargo.lock from this PR?" ;)

**Pull request closed without merging**